wrappers/src/marshalling.hpp

```diff
@@ -86,6 +95,12 @@ struct MarshaledVector
 
     iterator begin() const noexcept { return {items}; }
     iterator end() const noexcept { return {items + count}; }
+
+    /// Needed for MSVC when returning a `MarshaledVector` from CPP
+    /// directly, as compared to when nested within another struct.
+    AnyMarshaledVector for_any_type_marshalling() const {
+        return {items, count};
+    }
```

Comment:
Note sure this brings in a lot of value vs just creating `AnyMarshaledVector` directly. I feel 

```cpp
REALM_EXPORT AnyMarshaledVector shared_realm_get_log_category_names() {
  // ...
  return { result.data(), result.size() };
}
```

reads better than

```cpp
REALM_EXPORT AnyMarshaledVector shared_realm_get_log_category_names() {
  // ...
  return MarshaledVector(result).for_any_type_marshalling();
}
```

Reply:
It was mainly to follow the pattern used in e.g. [error_handling.hpp](https://github.com/realm/realm-dotnet/blob/4bc30579d5d11ecfab52cb5452423af4f31fc686/wrappers/src/error_handling.hpp#L75), although that code has additional logic prior to returning the struct.